goaviatrix/aws_tgw_vpc_attachment.go

```diff
@@ -381,3 +390,47 @@ func (c *Client) EditTgwSpokeVpcCustomizedRouteAdvertisement(awsTgwVpcAttachment
 	}
 	return nil
 }
+
+func (c *Client) UpdateFirewallAttachmentAccessFromOnprem(awsTgwVpcAttachment *AwsTgwVpcAttachment) error {
+	params := map[string]string{
+		"action":          "update_firewall_attachment_access_from_onprem",
+		"CID":             c.CID,
+		"tgw_name":        awsTgwVpcAttachment.TgwName,
+		"attachment_name": awsTgwVpcAttachment.VpcID,
+		"edge_attachment": awsTgwVpcAttachment.EdgeAttachment,
+	}
+
+	if awsTgwVpcAttachment.EdgeAttachment == "" {
+		params["edge_attachment"] = "no"
+	}
+
+	return c.PostAPI(params["action"], params, BasicCheck)
+}
+
+func (c *Client) GetFirenetManagementDetails(awsTgwVpcAttachment *AwsTgwVpcAttachment) ([]string, error) {
+	params := map[string]string{
+		"action":          "get_tgw_attachment_details",
+		"CID":             c.CID,
+		"tgw_name":        awsTgwVpcAttachment.TgwName,
+		"attachment_name": awsTgwVpcAttachment.VpcID,
+	}
+
+	type AccessFromEdge struct {
+		AccessFromEdge []string `json:"access_from_edge"`
+	}
+
+	type Resp struct {
+		Return  bool             `json:"return"`
+		Results []AccessFromEdge `json:"results"`
+		Reason  string           `json:"reason"`
+	}
+
+	var data Resp
+
+	err := c.GetAPI(&data, params["action"], params, BasicCheck)
+	if err != nil {
+		return nil, err
+	}
+
+	return data.Results[0].AccessFromEdge, nil
```

Comment:
Will `Results` never be empty? What if the attachment is deleted on the GUI, would this panic?

Reply:
If the vpc is not attached, then `access_from_edge` will not be in json.